go-sdk/internal/sparse-merkle-tree/smt/merkletree.go

```diff
@@ -249,23 +272,23 @@ func (mt *sparseMerkleTree) addLeaf(newLeaf core.Node, currentNodeIndex core.Nod
 // must be called from inside a write lock
 // addNode adds a node into the MT.  Empty nodes are not stored in the tree;
 // they are all the same and assumed to always exist.
-func (mt *sparseMerkleTree) addNode(n core.Node) (core.NodeIndex, error) {
+func (mt *sparseMerkleTree) addNode(batch core.Transaction, n core.Node) (core.NodeIndex, error) {
 	if n.Type() == core.NodeTypeEmpty {
 		return n.Ref(), nil
 	}
 	k := n.Ref()
 	// Check that the node key doesn't already exist
-	if _, err := mt.db.GetNode(k); err == nil {
+	if _, err := batch.GetNode(k); err == nil {
 		return nil, ErrNodeIndexAlreadyExists
 	}
-	err := mt.db.InsertNode(n)
+	err := batch.InsertNode(n)
```

Comment:
This is where I don't understand the batch function where you get a node but the TX is not commit yet and then you are adding an insert to same transaction

Reply:
for example here, you would use db instead of batch for GetNode?